### PR TITLE
Fix reconciler TXT record handling

### LINKED ISSUE #462: TXT record values must be wrapped in "...", split at 255 chars, etc.
See https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/ResourceRecordTypes.html#TXTFormat

We fought with Route53 all night tonight to get TXT records to work.  It turns out we need to:

1. wrap the value in extra "..."
2. break it at the 255 character mark
3. not exceed 4,000 characters

[edit]
- break it at the 255 character mark
- escape backslash and quotation mark characters \ => \\  and " => \"
- escape not \o40 - \o177 ascii characters in 3 digit octal format " " => "\040"
- wrap the chunks in extra "..."
- join them with a space

### PR DESCRIPTION
**Pull request by dadolhay**
Closes #462

### CODE REVIEW

**Review by humphd — CHANGES_REQUESTED** [review]
Please add a few unit tests for these 2 utility functions so we don't waste time debugging this on staging.

**Comment by dadolhay** [comment]
@humphd Added!

File: app/queues/reconciler/route53Utils.server.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+  // Create an uninitialized array with the length to hold our split up strings (max 255 chars)
+  const segments = new Array(Math.ceil(value.length / 255))
+    // Initialize with undefined
+    .fill(undefined)
+    // Loop through, using the index split out the appropriate parts from the original string
+    .map((segment, index) => value.substring(index * 255, (index + 1) * 255))
+    // escape
+    .map((segment) => segment.replace(/([^!-~]|[\\"])/g, escapeFn))
+    // add parenthesis around the segments
```
Comment (humphd) [inline app/queues/reconciler/route53Utils.server.ts:55]: parenthesis?  Do you mean "double-quotes"?

**Review by humphd — DISMISSED** [review]
Excellent, thank you.  The only thing this doesn't do is the 4,000 limit:

> The maximum length of a value in a TXT record is 4,000 characters.

However, I'm OK if we leave that out.

**Comment by dadolhay** [comment]
> Excellent, thank you. The only thing this doesn't do is the 4,000 limit:
> 
> > The maximum length of a value in a TXT record is 4,000 characters.
> 
> However, I'm OK if we leave that out.

Actually, I don't think our db column holds that amount of info. Even if it did, we should inhibit the insert IMO

File: app/queues/reconciler/route53Utils.server.ts (hunk 1 above)
Comment (dadolhay) [reply app/queues/reconciler/route53Utils.server.ts]: yep! thanks!

**Review by humphd — APPROVED** [review]

**Review by Eakam1007 — APPROVED** [review]

**Pull request merged**